### PR TITLE
`azurerm_app_service` - support for MSI

### PR DESCRIPTION
**Pull request by sabbox**
This PR enables Managed Service Identity for App Service #1106
- create new service with MSI
- update existed one with MSI

### CODE REVIEW

File: azurerm/resource_arm_app_service.go [hunk 1]
```diff
@@ -29,6 +29,33 @@ func resourceArmAppService() *schema.Resource {
 				ValidateFunc: validateAppServiceName,
 			},
 
+			"identity": {
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_app_service.go]: Can we document these properties in the documentation? That's stored in this file, it'll want the `type` property documented in the `Argument Reference` block and the `principal_id` and `tenant_id` fields documented in the `Attributes Reference` block

File: azurerm/resource_arm_app_service.go [hunk 2]
```diff
@@ -431,6 +473,24 @@ func resourceArmAppServiceUpdate(d *schema.ResourceData, meta interface{}) error
...
+
+		site, err := client.Get(ctx, resGroup, name)
+		if err != nil {
+			return fmt.Errorf("Error getting configuration for App Service %q: %+v", name, err)
+		}
+
+		appServiceIdentity := expandAzureRmAppServiceIdentity(d)
+		site.Identity = appServiceIdentity
+
+		future, err := client.CreateOrUpdate(ctx, resGroup, name, site)
```
[the remark this answers is not in the document]
Comment (sabbox) [reply azurerm/resource_arm_app_service.go]: Sure, sorry my bad

File: azurerm/resource_arm_app_service.go (hunk 1 above)
Comment (sabbox) [reply azurerm/resource_arm_app_service.go]: done

**Comment by tombuildsstuff** [comment]
Tests pass:

screen shot 2018-05-01 at 15 43 38

**Pull request merged**